### PR TITLE
vmware_vm_facts: pep8, imports, docs, cosmetics

### PR DESCRIPTION
**Pull request by pchauncey**
##### SUMMARY

- PEP8 compliance
- Cosmetics
- AnsibleModule import

##### ISSUE TYPE
 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME
vmware_vm_facts module

##### ANSIBLE VERSION
```
ansible 2.4.0 (vmware_vm_facts-pep8 a6bccb286d) last updated 2017/05/19 12:03:07 (GMT -500)
```

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_vm_facts.py [hunk 1]
```diff
@@ -85,7 +93,8 @@ def get_all_virtual_machines(content):
 def main():
 
     argument_spec = vmware_argument_spec()
-    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=False)
```
Comment (dagwieers) [inline lib/ansible/modules/cloud/vmware/vmware_vm_facts.py]: This is not strictly necessary, as lines can have 160 columns.

File: lib/ansible/modules/cloud/vmware/vmware_vm_facts.py [hunk 2]
```diff
@@ -101,8 +110,9 @@ def main():
     except Exception as e:
         module.fail_json(msg=str(e))
 
-from ansible.module_utils.vmware import *
-from ansible.module_utils.basic import *
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import (connect_to_api, get_all_objs,
+                                         vmware_argument_spec)
```
Comment (dagwieers) [inline lib/ansible/modules/cloud/vmware/vmware_vm_facts.py]: Imports are now specified right after the documentation strings.

**Review by dagwieers — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_vm_facts.py [hunk 3]
```diff
@@ -23,14 +23,14 @@
...
-DOCUMENTATION = '''
+DOCUMENTATION = r'''
 ---
 module: vmware_vm_facts
-short_description: Return basic facts pertaining to a vSphere virtual machine guest
+short_description: Return basic facts pertaining to vSphere virtual machines.
 description:
-    - Return basic facts pertaining to a vSphere virtual machine guest
+    - Return basic facts pertaining to vSphere virtual machines.
 version_added: 2.0
```
Comment (dagwieers) [inline lib/ansible/modules/cloud/vmware/vmware_vm_facts.py]: version_added needs to be a string.

**Review by dagwieers — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_vm_facts.py (hunk 3 above)
Comment (pchauncey) [reply lib/ansible/modules/cloud/vmware/vmware_vm_facts.py]: Thanks appreciate it.  Update pushed.

**Comment by pchauncey** [comment]
@dagwieers  Thanks for the suggested changes, done.

ready_for_review

**Comment by ryansb** [comment]
!needs_revision

**Review by Akasurde — APPROVED** [review]
shipit

**Comment by pchauncey** [comment]
bot_status

**Comment by pchauncey** [comment]
ready_for_review

File: lib/ansible/modules/cloud/vmware/vmware_vm_facts.py [hunk 4]
```diff
@@ -23,14 +23,14 @@
                     'supported_by': 'community'}
 
 
-DOCUMENTATION = '''
+DOCUMENTATION = r'''
 ---
 module: vmware_vm_facts
-short_description: Return basic facts pertaining to a vSphere virtual machine guest
+short_description: Return basic facts pertaining to vSphere virtual machines.
```
Comment (dagwieers) [inline lib/ansible/modules/cloud/vmware/vmware_vm_facts.py]: Short description should not have trailing dot. (Other descriptions do...)

File: lib/ansible/modules/cloud/vmware/vmware_vm_facts.py [hunk 5]
```diff
@@ -23,14 +23,14 @@
...
 description:
-    - Return basic facts pertaining to a vSphere virtual machine guest
-version_added: 2.0
-author: "Joseph Callen (@jcpowermac)"
+    - Return basic facts pertaining to vSphere virtual machines.
+version_added: "2.0"
+author: Joseph Callen (@jcpowermac)
```
Comment (dagwieers) [inline lib/ansible/modules/cloud/vmware/vmware_vm_facts.py]: Make this a proper YAML list:
```yaml
author:
- Joseph Callen (@jcpowermac)
```

**Review by dagwieers — CHANGES_REQUESTED** [review]
LGTM only if we have integration tests for this module. Rewriting imports is dangerous without proper integration tests.

**Comment by pchauncey** [comment]
@dagwieers Thanks, I'll pull the import cleanup changes for now and pending some free cycles get some integration test written too.  Other changes implemented as well.

**Pull request closed without merging**